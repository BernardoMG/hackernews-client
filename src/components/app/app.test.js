/* eslint-disable no-undef */
import React from 'react'
import ReactDOM from 'react-dom'
import Renderer from 'react-test-renderer'
import App from './app.js'

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<App />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('has a valid snapshot', () => {
    const component = Renderer.create(<App /> )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

