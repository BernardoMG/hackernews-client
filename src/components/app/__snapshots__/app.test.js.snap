// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="page"
>
  <div
    className="interactions"
  >
    <form
      onSubmit={[Function]}
    >
      Search
      <input
        onChange={[Function]}
        type="text"
        value="redux"
      />
      <button
        className="button"
        onClick={null}
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
  <div
    className="table"
  >
    <div
      className="table-header"
    >
      <span
        style={
          Object {
            "width": "40%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
        >
          Title
        </button>
      </span>
      <span
        style={
          Object {
            "width": "30%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
        >
          Author
        </button>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
        >
          Comments
        </button>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        <button
          className="button-inline"
          onClick={[Function]}
        >
          Points
        </button>
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        Archive
      </span>
    </div>
  </div>
  <div
    className="interactions"
  >
    <div>
      Loading ...
    </div>
  </div>
</div>
`;
